test/powershell/Modules/Microsoft.PowerShell.Management/Get-Item.Tests.ps1

```diff
@@ -17,7 +25,89 @@ Describe "Get-Item" -Tags "CI" {
     }
+
+    It "Should not return the item unless force is used if hidden" {
+        ${hiddenFile} = "${TESTDRIVE}/.hidden.txt"
+        ${item} = New-Item -type file "${hiddenFile}"
+        if ( ${IsWindows} ) {
+            attrib +h "$hiddenFile"
+        }
+        ${result} = Get-Item "${hiddenFile}" -ErrorAction SilentlyContinue
+        ${result} | Should BeNullOrEmpty
+        ${result} = Get-Item -force "${hiddenFile}" -ErrorAction SilentlyContinue
+        ${result}.FullName | Should Be ${item}.FullName
+    }
+
+    Context "Test for Include, Exclude, and Filter" {
+        BeforeAll {
+            ${testBaseDir} = "${TESTDRIVE}/IncludeExclude"
+            $null = New-Item -Type Directory "${testBaseDir}"
+            $null = New-Item -Type File "${testBaseDir}/file1.txt"
+            $null = New-Item -Type File "${testBaseDir}/file2.txt"
+        }
+        It "Should respect -Exclude" {
+            $result = Get-Item "${testBaseDir}/*" -Exclude "file2.txt"
+            ($result).Count | Should Be 1
+            $result.Name | should be "file1.txt"
+        }
+        It "Should respect -Include" {
+            $result = Get-Item "${testBaseDir}/*" -Include "file2.txt"
+            ($result).Count | Should Be 1
+            $result.Name | should be "file2.txt"
+        }
+        It "Should respect -Filter" {
+            $result = Get-Item "${testBaseDir}/*" -Filter "*2*"
+            ($result).Count | Should Be 1
+            $result.Name | should be "file2.txt"
+        }
+    }
+
+    Context "Error Condition Checking" {
+        It "Should return an error if the provider does not exist" {
+            { Get-Item BadProvider::/BadFile -ErrorAction Stop } | ShouldBeErrorId "ProviderNotFound,Microsoft.PowerShell.Commands.GetItemCommand"
+        }
+
+        It "Should return an error if the drive does not exist" {
+            { Get-Item BadDrive:/BadFile -ErrorAction Stop } | ShouldBeErrorId "DriveNotFound,Microsoft.PowerShell.Commands.GetItemCommand"
+        }
+    }
+
+    Context "Alternate Stream Tests" {
+        BeforeAll {
+            if ( $skip )
+            {
+                return
+            }
+            $altStreamPath = "$TESTDRIVE/altStream.txt"
+            $stringData = "test data"
+            $streamName = "test"
+            $item = new-item -type file $altStreamPath
+            Set-Content -path $altStreamPath -Stream $streamName -Value $stringData
+        }
+        It "Should find an alternate stream if present" -skip:$skip {
+            $result = Get-Item $altStreamPath -Stream $streamName
+            $result.Length | Should Be ($stringData.Length + [Environment]::NewLine.Length)
+            $result.Stream | Should Be $streamName
+        }
+
+    }
+
+    Context "Registry Provider" {
+        It "Can retrieve an item from registry" -skip:$skip {
+            ${result} = Get-Item HKLM:/Software
+            ${result} | Should BeOfType "Microsoft.Win32.RegistryKey"
+        }
```

Comment:
function, alias and environment provider also have similar gaps. Either add tests is this PR or a new one.